saleor/graphql/shop/mutations.py

```diff
@@ -288,3 +288,78 @@ def perform_mutation(cls, _root, info, key_type):
 
         instance.delete()
         return AuthorizationKeyDelete(authorization_key=instance, shop=Shop())
+
+
+class StaffNotificationInput(graphene.InputObjectType):
+    staff_email = graphene.String(
+        required=False,
+        description="Staff email (instead of providing staff member user).",
+    )
+    active = graphene.Boolean(required=False, description="Client secret.")
+    user = graphene.ID(required=False, description="The ID of the user to assign.")
+
+
+class StaffNotificationCreate(ModelMutation):
+    class Arguments:
+        input = StaffNotificationInput(
+            required=True, description="Fields required to update a product."
+        )
+
+    class Meta:
+        description = "Creates a new category."
+        model = account_models.StaffNotification
+        permissions = ("product.manage_products",)
+        error_type_class = ShopError
+        error_type_field = "shop_errors"
+
+    @classmethod
+    def clean_input(cls, info, instance, data):
+        cleaned_input = super().clean_input(info, instance, data)
+        if "staff_email" not in cleaned_input and "user" not in cleaned_input:
+            raise ValidationError(
+                {
+                    "staff_notification": ValidationError(
+                        "User or staff email is required", code=ShopErrorCode.REQUIRED
+                    )
+                }
+            )
+        cleaned_input["site_settings"] = info.context.site.settings
+        cleaned_input["active"] = cleaned_input.get("active", True)
+        if "staff_email" in cleaned_input:
+            staff_email = cleaned_input["staff_email"]
+            staff_user = account_models.User.objects.filter(email=staff_email).first()
+            if staff_user:
+                cleaned_input["user"] = staff_user
+                cleaned_input["staff_email"] = None
+        return cleaned_input
+
+
+class StaffNotificationUpdate(StaffNotificationCreate):
+    class Arguments:
+        id = graphene.ID(
+            required=True, description="ID of a StaffNotification to update."
```

Comment:
We don't have to reveal internal model names in descriptions since it goes to the API documentation later. 
```suggestion
            required=True, description="ID of a staff notification to update."
```